### DISCUSSION TITLE
extending model (site genesis)

### DISCUSSION THREAD

**Discussion by micheleezza** [SalesforceCommerceCloud/sfcc-ci#463] [Team Posts]
Hi guys, I'm new to Salesforce CC developing.
At the moment I'm developing a link cartridge for SFCC using sitegenesis technology.
I've to add a custom action after the "order.placeOrder" method in the OrderModel.js model.

What is the best method to extend a model in the storefront_controllers cartridge?
I saw that the app class calls only model in its directory so do I have to move the app class into my cartridge too? It sounds not good to me.

thanks to all
Michele

**Comment by Ivan-Melnyk** [comment]
Hi @micheleezza ,
the best way of extending anything in SFCC without changing original files will be to have the separate cartridge and, depending on the architecture and version of models, controllers, etc., - either extend the functionality of the file or replace it.
Please have a look at module.superModule of the common JS, you can find a lot of examples how to use it in SFRA. To extend the class (function) you can try to use prototype, to extend the object - you can try to use Object.create(originalObject, {your custom methods and properties}).

**Comment by micheleezza** [comment]
Thanks Ivan
For  SiteGenesis, I can't use superModule, can I?

**Comment by Ivan-Melnyk** [comment]
You can, since regardless of is it SFRA or SiteGenesis - the version of common JS is the same on SFCC Digital.
Please also check the compatibility mode of your SFCC Digital.

**Comment by hnestmann** [comment]
superModule is part of the API so it works with SG too. Howeve customers on SiteGenesis often change the code, I would also write a summary what you overload does and how to integrate it if OrderModel was changed   or moved

**Comment by micheleezza** [comment]
Ok I've understood.
But now the second problem: the controllers of sitegenesis calls "app.js" module to load the models. And its getModel method:

exports.getModel = function (modelName) {
    return require('./models/' + modelName + 'Model');
};

calls only models that are in the same directory. So how can I let SF use my model extension? Do I have to extend also app module too?

**Comment by Ivan-Melnyk** [comment]
You can make a copy the original app.js into your extension cartridge and slightly modify it, use "*/..." instead of "./..." (according to common JS). And yes, SiteGenesis architecture is not that flexible as SFRA in regards of customization.
